Loop/View Controllers/InsulinDeliveryTableViewController.swift

```diff
@@ -8,6 +8,520 @@
+
+                tableView.deleteRows(at: [indexPath], with: .automatic)
+
+                doseStore?.deleteReservoirValue(value) { (_, error) -> Void in
+                    if let error = error {
+                        DispatchQueue.main.async {
+                            self.present(UIAlertController(with: error), animated: true)
+                            self.reloadData()
+                        }
+                    }
+                }
+            case .history(let historyValues):
+                var historyValues = historyValues
+                let value = historyValues.remove(at: indexPath.row)
+                self.values = .history(historyValues)
+
+                tableView.deleteRows(at: [indexPath], with: .automatic)
+
+                doseStore?.deletePumpEvent(value) { (error) -> Void in
+                    if let error = error {
+                        DispatchQueue.main.async {
+                            self.present(UIAlertController(with: error), animated: true)
+                            self.reloadData()
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if case .display = state, case .history(let history) = values {
+            let entry = history[indexPath.row]
+
+            let vc = CommandResponseViewController(command: { (completionHandler) -> String in
+                var description = [String]()
+
+                description.append(self.timeFormatter.string(from: entry.date))
+
+                if let title = entry.title {
+                    description.append(title)
+                }
+
+                if let dose = entry.dose {
+                    description.append(String(describing: dose))
+                }
+
+                if let raw = entry.raw {
+                    description.append(raw.hexadecimalString)
+                }
+
+                return description.joined(separator: "\n\n")
+            })
+
+            vc.title = NSLocalizedString("Pump Event", comment: "The title of the screen displaying a pump event")
+
+            show(vc, sender: indexPath)
+        }
+    }
+
+}
+
+fileprivate extension UIAlertController {
+    convenience init(deleteAllConfirmationMessage: String, confirmationHandler handler: @escaping () -> Void) {
+        self.init(
+            title: nil,
+            message: deleteAllConfirmationMessage,
+            preferredStyle: .actionSheet
+        )
+        
+        addAction(UIAlertAction(
+            title: NSLocalizedString("Delete All", comment: "Button title to delete all objects"),
+            style: .destructive,
+            handler: { (_) in handler() }
+        ))
+
+        addAction(UIAlertAction(
+            title: NSLocalizedString("Cancel", comment: "The title of the cancel action in an action sheet"),
+            style: .cancel
+        ))
+    }
+}
```

Comment:
Didn't review this file extensively; I'm assuming it's copied from LoopKit aside from the instantiation of the bolus view controller near the top of the file.

Reply:
Yes, there were only minor changes (adding the "+" symbol, instantiating the view controller, and adding the doseStore variable)